src/Controller/Controller.php

```diff
@@ -124,6 +126,7 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      *
      * @var \Cake\Http\ServerRequest
      * @link https://book.cakephp.org/3.0/en/controllers/request-response.html#request
+     * @deprecated 3.5.0 Use getRequest()/setRequest instead.
```

Comment:
@ionas I prefer the consistency. All getters here have a prefix, so this one should also have it. BTW `setRequest` has been around for the while and is used to set the state of the controller.